llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -646,7 +650,9 @@ Value *VPInstruction::generatePerPart(VPTransformState &State, unsigned Part) {
            "can only generate first lane for PtrAdd");
     Value *Ptr = State.get(getOperand(0), Part, /* IsScalar */ true);
     Value *Addend = State.get(getOperand(1), Part, /* IsScalar */ true);
-    return Builder.CreatePtrAdd(Ptr, Addend, Name);
+    return Builder.CreatePtrAdd(Ptr, Addend, Name,
+                                isInBounds() ? GEPNoWrapFlags::inBounds()
+                                             : GEPNoWrapFlags::none());
```

Comment:
Have 
```
GEPNoWrapFlags VPRecipeWithIRFlags::getGEPNoWrapFlags() const;
```
similar to
```
FastMathFlags VPRecipeWithIRFlags::getFastMathFlags() const
```
and then
```suggestion
    return Builder.CreatePtrAdd(Ptr, Addend, Name, getGEPNoWrapFlags());
```
?